ethcore/src/miner/miner.rs

```diff
@@ -117,90 +117,89 @@ impl Miner {
 		//   otherwise, leave everything alone.
 		// otherwise, author a fresh block.
 */
-
-		let (b, invalid_transactions) = match sealing_work.pop_if(|b| b.block().fields().header.parent_hash() == &best_hash) {
+		let mut open_block = match sealing_work.pop_if(|b| b.block().fields().header.parent_hash() == &best_hash) {
 			Some(old_block) => {
 				trace!(target: "miner", "Already have previous work; updating and returning");
 				// add transactions to old_block
 				let e = self.engine();
-				let mut invalid_transactions = HashSet::new();
-				let mut block = old_block.reopen(e, chain.vm_factory());
-				let block_number = block.block().fields().header.number();
-
-				// TODO: push new uncles, too.
-				// TODO: refactor with chain.prepare_sealing
-				for tx in transactions {
-					let hash = tx.hash();
-					let res = block.push_transaction(tx, None);
-					match res {
-						Err(Error::Execution(ExecutionError::BlockGasLimitReached { gas_limit, gas_used, .. })) => {
-							trace!(target: "miner", "Skipping adding transaction to block because of gas limit: {:?}", hash);
-							// Exit early if gas left is smaller then min_tx_gas
-							let min_tx_gas: U256 = 21000.into();	// TODO: figure this out properly.
-							if gas_limit - gas_used < min_tx_gas {
-								break;
-							}
-						},
-						Err(Error::Transaction(TransactionError::AlreadyImported)) => {}	// already have transaction - ignore
-						Err(e) => {
-							invalid_transactions.insert(hash);
-							trace!(target: "miner",
-								   "Error adding transaction to block: number={}. transaction_hash={:?}, Error: {:?}",
-								   block_number, hash, e);
-						},
-						_ => {}	// imported ok
-					}
-				}
-				(Some(block.close()), invalid_transactions)
+				old_block.reopen(e, chain.vm_factory())
 			}
 			None => {
 				// block not found - create it.
 				trace!(target: "miner", "No existing work - making new block");
-				chain.prepare_sealing(
+				chain.prepare_open_block(
 					self.author(),
 					self.gas_floor_target(),
-					self.extra_data(),
-					transactions,
+					self.extra_data()
 				)
 			}
 		};
+
+		let mut invalid_transactions = HashSet::new();
```

Comment:
the transactions are no longer added in `prepare_open_block` (né `prepare_sealing`)?